addons/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/handler/welcome/AbstractNetatmoWelcomeHandler.java

```diff
@@ -0,0 +1,192 @@
+/**
+ * Copyright (c) 2014-2016 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.netatmo.handler.welcome;
+
+import static org.openhab.binding.netatmo.NetatmoBindingConstants.*;
+
+import java.io.IOException;
+import java.util.HashMap;
+
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.netatmo.handler.NetatmoBridgeHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParser;
+import com.squareup.okhttp.OkHttpClient;
+import com.squareup.okhttp.Request;
+import com.squareup.okhttp.Response;
+
+import io.swagger.client.model.NAWelcomeCameras;
+import io.swagger.client.model.NAWelcomeHomes;
+
+/**
+ * {@link AbstractNetatmoWelcomeHandler} is the abstract class that handles
+ * common behaviors of both Devices and Modules
+ *
+ * @author Ing. Peter Weiss - Welcome camera implementation
+ *
+ */
+abstract class AbstractNetatmoWelcomeHandler extends BaseThingHandler {
```

Comment:
There are many informations in the class AbstractNetatmoThingHandler which will not be used in the welcome things, like signalTresholds, measuredChannels an measures.
If the Abstract class just holds all generic informations for all netatmo things, it would be a good idea to inherit from there.